modules/product/src/main/java/com/opengamma/strata/product/swap/type/FixedInflationSwapConvention.java

```diff
@@ -0,0 +1,212 @@
+   */
+  public abstract FixedRateSwapLegConvention getFixedLeg();
+
+  /**
+   * Gets the market convention of the floating leg.
+   * 
+   * @return the floating leg convention
+   */
+  public abstract InflationRateSwapLegConvention getFloatingLeg();
+
+  /**
+   * Gets the offset of the spot value date from the trade date.
+   * <p>
+   * The offset is applied to the trade date to find the start date.
+   * A typical value is "plus 2 business days".
+   * 
+   * @return the spot date offset, not null
+   */
+  public abstract BusinessDayAdjustment getSpotDateOffset();
+  
+  /**
+   * Gets the offset of the payment date.
+   * 
+   * @return the spot date offset, not null
+   */
+  public abstract DaysAdjustment getPaymentDateOffset();
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates a forward-starting trade based on this convention.
+   * <p>
+   * The notional is unsigned, with buy/sell determining the direction of the trade.
+   * If buying the swap, the floating rate is received from the counterparty, with the fixed rate being paid.
+   * If selling the swap, the floating rate is paid to the counterparty, with the fixed rate being received.
+   * 
+   * @param tradeDate  the date of the trade
+   * @param tenor  the tenor of the trade
+   * @param lag       the positive period between the price index and the accrual date, typically a number of months.
+   * @param buySell  the buy/sell flag
+   * @param notional  the notional amount
+   * @param fixedRate  the fixed rate, typically derived from the market
+   * @param refData  the reference data, used to resolve the trade dates
+   * @return the trade
+   * @throws ReferenceDataNotFoundException if an identifier cannot be resolved in the reference data
+   */
+  public default SwapTrade createTrade(
+      LocalDate tradeDate,
+      Tenor tenor,
+      Period lag,
+      BuySell buySell,
+      double notional,
+      double fixedRate,
+      ReferenceData refData) {
+
+    LocalDate spotValue = calculateSpotDateFromTradeDate(tradeDate, refData);
+    LocalDate startDate = spotValue.plus(Period.ZERO);
+    LocalDate endDate = startDate.plus(tenor.getPeriod());
+    return toTrade(tradeDate, startDate, endDate, lag, buySell, notional, fixedRate);
+  }
+
+  /**
+   * Creates a trade based on this convention.
+   * <p>
+   * This returns a trade based on the specified dates.
+   * <p>
+   * The notional is unsigned, with buy/sell determining the direction of the trade.
+   * If buying the swap, the floating rate is received from the counterparty, with the fixed rate being paid.
+   * If selling the swap, the floating rate is paid to the counterparty, with the fixed rate being received.
+   * 
+   * @param tradeDate  the date of the trade
+   * @param startDate  the start date
+   * @param endDate  the end date
+   * @param lag       the positive period between the price index and the accrual date, typically a number of months.
```

Comment:
Formatting - too many spaces between `lag` and the description. Same problem below.